watertap/examples/SOP/demo_SOP.py

```diff
@@ -0,0 +1,132 @@
+from pyomo.environ import (
```

Comment:
I am not sure how necessary this file is since it is essentially a copy of the test file. Maybe it could subsequently be built upon, e.g., by adding a feed pump at the beginning. We should consider whether we want to remove this file from this PR or followup with another PR to bolster this example. Not critical though

Reply:
A trivial comment: I think we should extend file names to `selective_oil_permeation` rather than SOP

Reply:
I am on board with removing `demo_SOP.py` and using "selective_oil_permeation" instead of "SOP" in all names.

Do you think we should also get rid of `demo_SOP_prop_pack.py`?

Reply:
I think so. If we are going to have an example, I think it should be a flowsheet which makes use of the prop model as well as other units, such as pumps, as well as a Feed and Product block. Alternatively, we could morph these demo files into something on the documentation side, serving as a how-to or tutorial perhaps

Reply:
But that could be for a future PR, if we decide to delve into this further. For this PR, I'd say remove both demo files. You should still be able to access them if ever you wanted to revamp them.

Reply:
Addressed in latest push